spec/models/api_key_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ApiKey, type: :model do
+  it { is_expected.to belong_to(:organization) }
+
+  describe '#save' do
+    subject { api_key.save! }
+
+    before do
+      allow(api_key).to receive(:set_value).and_call_original
+      subject
+    end
+
+    context 'with a new record' do
+      let(:api_key) { build(:api_key) }
+
+      it 'calls #set_value' do
+        expect(api_key).to have_received(:set_value)
+      end
+    end
+
+    context 'with a persisted record' do
+      let(:api_key) { create(:api_key) }
+
+      it 'does not call #set_value' do
+        expect(api_key).not_to have_received(:set_value)
+      end
+    end
```

Comment:
I would avoid stubbing when possible...
here you can just check an uuid is assigned to a new records when saved,
it does not change when the record is already persisted and have one assigned...
and it handles the collision with an existing record with the same value (here you will have to stub SecureRandom I guess).